File: frontend/src/types/comment.ts
export interface CommentItem {
  id: string;
  document_id: string;
  version_id: string | null;
  user_id: string;
  content: string;
  page_number: number | null;
  is_resolved: boolean;
  created_at: string | null;
  user?: {
    id: string;
    full_name: string;
    role: string;
  };
}

export interface CommentCreatePayload {
  content: string;
  version_id?: string;
  page_number?: number;
}

export interface CommentUpdatePayload {
  content: string;
}

File: app/db/models/comment.py
from datetime import datetime
from uuid import UUID, uuid4

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base_class import Base


class Comment(Base):
    __tablename__ = "comments"

    id: Mapped[UUID] = mapped_column(primary_key=True, default=uuid4)
    document_id: Mapped[UUID] = mapped_column(ForeignKey("documents.id"), index=True)
    version_id: Mapped[UUID | None] = mapped_column(ForeignKey("document_versions.id"), nullable=True)
    user_id: Mapped[UUID] = mapped_column(ForeignKey("users.id"), index=True)
    content: Mapped[str] = mapped_column(Text)
    page_number: Mapped[int | None] = mapped_column(Integer, nullable=True)
    is_resolved: Mapped[bool] = mapped_column(Boolean, default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )
    deleted_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
